spec/javascript/components/molecules/StatusDropdown.test.js

```diff
@@ -0,0 +1,130 @@
+  test('it renders a select', () => {
+    wrapper = getWrapper()
+    expect(wrapper.find('Select').exists()).toBeTruthy()
+  })
+
+  test('it renders with default props correctly', () => {
+    wrapper = getWrapper()
+
+    const dropdownProps = wrapper.find('Dropdown').props()
+    expect(dropdownProps.items).toEqual(LEASE_UP_STATUS_OPTIONS)
+    expect(dropdownProps.value).toBeNull()
+    expect(dropdownProps.placeholder).toEqual('Status')
+    expect(dropdownProps.disabled).toBeFalsy()
+  })
+
+  describe('it renders the toggle as expected', () => {
+    test('when value is null', () => {
+      wrapper = getWrapper()
+      const expectedButtonClasses = [
+        'button',
+        'dropdown-button',
+        'has-icon--right',
+        'text-align-left',
+        'tertiary'
+      ]
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.text()).toEqual('Status')
+      expect(toggleButton.hasClass(expectedButtonClasses.join(' '))).toEqual(true)
+      expect(toggleButton.prop('disabled')).toEqual(false)
+    })
+
+    test('without any additional button styles', () => {
+      wrapper = getWrapper()
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.hasClass('expand')).toBe(false)
+      expect(toggleButton.hasClass('tiny')).toBe(false)
+      expect(toggleButton.hasClass('small')).toBe(false)
+    })
+
+    test('when additional styles are provided', () => {
+      wrapper = getWrapper({ expand: true, size: 'tiny' })
+      let toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.hasClass('expand')).toBe(true)
+      expect(toggleButton.hasClass('tiny')).toBe(true)
+      expect(toggleButton.hasClass('small')).toBe(false)
+
+      wrapper = getWrapper({ expand: true, size: 'small' })
+      toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.hasClass('tiny')).toBe(false)
+      expect(toggleButton.hasClass('small')).toBe(true)
+    })
+
+    test('when value provided and is styled', () => {
+      wrapper = getWrapper({ status: 'Appealed' })
+      const expectedButtonClasses = [
+        'button',
+        'dropdown-button',
+        'has-icon--right',
+        'text-align-left',
+        'is-appealed'
+      ]
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.text()).toEqual('Appealed')
+      expect(toggleButton.hasClass(expectedButtonClasses.join(' '))).toBe(true)
+    })
+
+    test('when the dropdown is disabled', () => {
+      wrapper = getWrapper({ disabled: true })
+      const toggleButton = wrapper.find('Select').find('.status-dropdown__control').find('button')
+      expect(toggleButton.prop('disabled')).toEqual(true)
+    })
+
+    test('when a placeholder is provided', () => {
```

Comment:
I think semantically you're naming some of these tests more like how a describe block would be named.

I think `test` blocks should be more like "should do X" or "does X when Y"

So this could be test('renders a placeholder when provided')

Reply:
updated in StatusDropdown and SubstatusDropdown